plugins/Actions/API.php

```diff
@@ -499,13 +500,38 @@ protected function doFilterPageDatatableSearch($callBackParameters, $table, $sea
      *
      * @param DataTable|DataTable\Simple|DataTable\Map $dataTable
      * @param bool $isPageTitleType Whether we are handling page title or regular URL
+     * @param int $idSite
+     * @param string $period
+     * @param string $date
      */
-    private function filterActionsDataTable($dataTable, $isPageTitleType)
+    private function filterActionsDataTable($dataTable, $isPageTitleType, $idSite, $period, $date)
     {
+
+        // Find the goals being used in the table - we only want to query totals for goals that are in the table
+        $goalConversionTotals = [];
+        $goalIds = [];
+        foreach ($dataTable->getRowsWithoutSummaryRow() as $row) {
+            if (isset($row[PiwikMetrics::INDEX_GOALS])) {
+                foreach ($row[PiwikMetrics::INDEX_GOALS] as $goalIdString => $metrics) {
+                    $goalIds[$goalIdString] = $goalIdString;
+                }
+            }
+        }
+
+        // For each goal in the table retrieve the period total conversions and add to the array
+        foreach ($goalIds as $idGoal => $g) {
+            $archive = Archive::build($idSite, $period, $date);
```

Comment:
Should that `Archive::build` include the segment? Otherwise we will compare segmented action data, with unsegmented conversion numbers. Or is that intended. If so it would be good to add a comment

Reply:
It should probably include the segment as we'll want to show the viewed before rate relative to the conversions that happened for the segment. I've included the segment in the archive::build call.